### PR TITLE
Self contained Travis build - work without Internet connection

### PR DESCRIPTION
**Pull request by axic**
Don't rely on the availability of https://ethereum.github.io/solc-bin/bin/list.js. First step towards #2.

### CODE REVIEW

**Comment by axic** [comment]
Testing needed! Any takers @redsquirrel @chriseth @Denton-L? :smiley:

**Comment by axic** [comment]
Do we actually need to include the ZIP file? Github has an option to download a branch as a ZIP file.

**Comment by axic** [comment]
Another possible way is to download soljson.js during `npm run build`, which seem to make more sense. Added it as a commit.

File: src/app.js [hunk 1]
```diff
@@ -415,6 +400,34 @@ var run = function () {
 
   $('.asmOutput button').click(function () { $(this).parent().find('pre').toggle(); });
 
+  // ----------------- version selector-------------
+
+  // var soljsonSources is provided by bin/list.js
+
+  var versionList = true;
+  try {
+    soljsonSources;
```
Comment (chriseth) [inline src/app.js]: `window.soljsonSource === undefined`? (or document, whatever the global context is)

File: package.json [hunk 2]
```diff
@@ -9,7 +9,7 @@
     "browser-test-remote-ie": "nightwatch --config nightwatch.js --env ie",
     "browser-test-remote-chrome": "nightwatch --config nightwatch.js --env chrome",
     "browser-test-remote-safari": "nightwatch --config nightwatch.js --env safari",
-    "build": "mkdir -p build; browserify src/index.js -o build/app.js",
+    "build": "mkdir -p build; rm soljson.js; wget https://ethereum.github.io/solc-bin/soljson.js; browserify src/index.js -o build/app.js",
```
Comment (chriseth) [inline package.json]: `rm -f`?

File: src/app.js (hunk 1 above)
Comment (axic) [inline src/app.js]:
```
missingVar === undefined
Uncaught ReferenceError: missingVar is not defined(…)(anonymous function) @ VM2543:1
```

vs.

```
window.missingVar === undefined
true
```

Weird. Did the first and had that issue.

File: ci/deploy_from_travis.sh [hunk 3]
```diff
@@ -10,6 +10,9 @@ git checkout --orphan gh-pages
 git rm --cached -r .
 echo "# Automatic build" > README.md
 echo "Built website from {$SHA}. See https://github.com/ethereum/browser-solidity/ for details." >> README.md
+echo "To use an offline copy, download browser-solidity-$SHA.zip." >> README.md
+# ZIP the whole directory
+zip -r browser-solidity-$SHA.zip .
```
Comment (chriseth) [inline ci/deploy_from_travis.sh]: We should exclude the .git directory

File: package.json (hunk 2 above)
Comment (chriseth) [inline package.json]: Is that a good thing to do in the build step? This way it always accesses the internet.

Comment (axic) [inline package.json]: True, maybe it makes more sense to have a separate call for it

**Comment by chriseth** [comment]
I just merged the change that uses the solc npm package. How did you plan that this one here works with the other one?

**Comment by axic** [comment]
@chriseth I don't think it is affected. Let me rebase it first.

**Pull request merged**